tools/flash_shelly.py

```diff
@@ -0,0 +1,499 @@
+from sys import argv, exit
+from os import path, remove
+from subprocess import Popen, PIPE
+from time import sleep
+from importlib import util
+
+"""
+#  Copyright (c) 2020 Andrew Blackburn & Deomid "rojer" Ryabkov
+#  All rights reserved
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This script will probe for any shelly device on the network and it will
+#  attempt to update them to the latest firmware version available.
+#  This script will not flash any firmware to a device that is not already on a
+#  version of this firmware, if you are looking to flash your device from stock
+#  or any other firmware please follow instructions here:
+#  https://github.com/mongoose-os-apps/shelly-homekit/wiki
+#
+#  Shelly HomeKit flashing script utility
+#  Usage: -{m|l|a|n|y|h} {hostname(s) optional}
+#   -m {homekit|revert|keep}   Script mode.
+#   -l                         List info of shelly device.
+#   -a                         Run against all the devices on the network.
+#   -n                         Do a dummy run through.
+#   -y                         Do not ask any confirmation to perform the flash.
+#   -V                         Force a particular version.
+#   -D                         Enable Debug Logging, you can increase logging with (-DD).")
+#   -h                         This help text.
+#
+#  usage: python3 flash_shelly.py -la
+#  usage: python3 flash_shelly.py shelly1-034FFF
+"""
+
+logging.basicConfig(format='%(message)s')
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+WHITE='\033[1m' # Bright White
+RED='\033[1;91m' # Bright Purple
+GREEN='\033[1;92m' # Bright Green
+YELLOW='\033[1;93m' # Bright Yellow
+BLUE='\033[1;94m' # Bright Blue
+PURPLE='\033[1;95m' # Bright Purple
+NC='\033[0m' # Normal Color
+
+arch = platform.system()
+
+if not importlib.util.find_spec("zeroconf"):
+  print('Installing zeroconf...')
+  pipe = Popen('pip3 install zeroconf', shell=True, stdout=PIPE).stdout
+  out = (pipe.read())
+  from zeroconf import ServiceBrowser, Zeroconf
+else:
+  from zeroconf import ServiceBrowser, Zeroconf
+
+if not importlib.util.find_spec("requests"):
+  print('Installing requests...')
+  pipe = Popen('pip3 install requests', shell=True, stdout=PIPE).stdout
+  out = (pipe.read())
+  import requests
+else:
+  import requests
+
+if not importlib.util.find_spec("packaging"):
+  print('Installing packaging...')
+  pipe = Popen('pip3 install packaging', shell=True, stdout=PIPE).stdout
```

Comment:
extra dependency, let's not use it. write python version of the js function instead.

Reply:
no it not that, I use it here 
```
    elif (version.parse(cfw) < version.parse(lfw)) and ((cfw_type == 'homekit' and mode == 'homekit') or (cfw_type == 'stock' and mode == 'stock') or mode == "keep"):
```

Reply:
done https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/5176422fad7ccf4ab1a55b93d76c7dea76b4cdaf